### PR TITLE
Accept EULA for all

### PR DESCRIPTION
**Pull request by majkinetor**

### CODE REVIEW

File: automatic/sysinternals/tools/chocolateyInstall.ps1 [hunk 1]
```diff
@@ -31,4 +31,7 @@ Install-ChocolateyZipPackage $packageName $url $installDir
   New-Item "$installDir\$_.exe.gui" -Type file -Force | Out-Null
 }
 
+"Accepting Eula for all applications"
+ls $installDir\*.exe | % { $k = "HKCU:/Software/Sysinternals/$($_.Name -replace '.exe$')"; mkdir $k -ea 0 -force; sp $k EulaAccepted 1 -ea 0}
```
Comment (ferventcoder) [inline automatic/sysinternals/tools/chocolateyInstall.ps1]: Wrap this with a try catch ignoring error. Or possibly a debug message with the error.

**Comment by majkinetor** [comment]
I don't believe above code can ever produce errors tbh because of -force and -ea
Ofc, in some funky rare cases maybe, but in my experience try {} catc{} doesn't always help with everything.

Anyway, .ike this? 
https://github.com/majkinetor/chocolatey-packages-1/blob/master/automatic/sysinternals/tools/chocolateyInstall.ps1

**Comment by dtgm** [comment]
Not all keys are the EXE name. You may want to hardcode registry keys into an array.

```
$keys = @(
'AccessChk',
'Active Directory Explorer',
'A',
'Autologon',
'AutoRuns',
'BGInfo',
'CacheSet',
'ClockRes',
'C',
'Coreinfo',
'Ctrl2cap',
'DbgView',
'Desktops',
'Disk2Vhd',
'DiskExt',
'Diskmon',
'DiskView',
'EFSDump',
'FindLinks',
'Handle',
'Hex2Dec',
'Junction',
'LdmDump',
'ListDLLs',
'LiveKd',
'LogonSessions',
'Movefile',
'PendMove',
'PipeList',
'Portmon',
'ProcDump',
'Process Explorer',
'Process Monitor',
'PsExec',
'psfile',
'PsGetSid',
'PsInfo',
'PsKill',
'PsList',
'PsLoggedon',
'PsLoglist',
'PsPasswd',
'PsPing',
'PsService',
'PsShutdown',
'PsSuspend',
'RamMap',
'RegDelNull',
'Regjump',
'RegSize',
'SDelete',
'Share Enum',
'ShellRunas - Sysinternals: www.sysinternals.com',
'SigCheck',
'Streams',
'Sync',
'TCPView',
'VMMap',
'VolumeID',
'Whois',
'Winobj',
'ZoomIt'
)
```

**Comment by michalzobec** [comment]
why you not use switch _-accepteula_ ? for example

`psexec.exe \\%remotemachine% -accepteula -u %remoteaccount% -p %remotepassword% -w %remoteshare%:\%remotedir% %remoteshare%:\%remotedir%\installontargetmachine.cmd`

**Comment by dtgm** [comment]
That would probably be better if it is incorporated into the shim generated. So something like:

```
$binFile = Join-Path "$(Split-Path -parent $MyInvocation.MyCommand.Definition)" "psexec.exe"
Install-BinFile -Name "psexec.exe" `
                -Path "$binFile" `
                -Command "-accepteula"
```

**Comment by dtgm** [comment]
Generate all shims:
```
$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
(Get-ChildItem $toolsDir\*.exe).name | ForEach-Object {
$binFile = Join-Path "$(Split-Path -parent $MyInvocation.MyCommand.Definition)" "$_"
Install-BinFile -Name "$_" `
                -Path "$binFile" `
                -Command "-accepteula"
```

**Comment by majkinetor** [comment]
Yeah, I suggested shim hardcoded parameters on the gitter but registry seemed easier if there was guaranties that exe names are used as keys. Since this is not correct, generating shims is far better idea.

**Comment by majkinetor** [comment]
That is not going to do it quite right because of `-UseStart` ? (shim option `-gui`).  You still need either hardcoding gui apps as it is now, or better, change Install-BinFile to not use -UseStart but look for the exe file subsystem via for example pscx Get-PEHeader:


    $guiApps = ls *.exe | ? { (Get-PEHeader $_).Subsystem -eq 'Windows' } | select -Expand Name

    $toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
    ls $toolsDir\*.exe | % {
        $params = {
            name    = (Split-Path -Leaf $_) -replace '.exe$'
            path    = $_.FullName
            command = '-accepteula'
        }
        $params.useStart = $guiApps -contains $_.name
        Install-BinFile @params
    }

I prefer the option to change `Install-BinFile` since then you don't have to care about anything - if sysinternals adds new tools, it will still work without package changes.

**Comment by ferventcoder** [comment]
> `$guiApps = ls *.exe | ? { (Get-PEHeader $_).Subsystem -eq 'Windows' } | select -Expand Name`

I'm interested in this bit. If this is mostly accurate, I think we can file an enhancement for shimgen

**Comment by majkinetor** [comment]
@ferventcoder 

I don't see how it can't be accurate. This is the relevant part:

https://github.com/mattifestation/PIC_Bindshell/blob/master/lib/PowerShell/Get-PEHeader.ps1#L255

(note: this is not the same Get-PeHeader function as that in pscx)

What you are interested in is parsing of

            WINDOWS_GUI =              2,   // Image runs in the Windows GUI subsystem.
            WINDOWS_CUI =              3,   // Image runs in the Windows character subsystem.

**Pull request closed without merging**